apps/meteor/app/apps/server/bridges/persistence.ts

```diff
@@ -125,6 +126,7 @@ export class AppPersistenceBridge extends PersistenceBridge {
 			associations,
 		};
 
-		return this.orch.getPersistenceModel().update(query, { $set: { data } }, { upsert });
+		// #TODO: #AppsEngineTypes - Remove explicit types and typecasts once the apps-engine definition/implementation mismatch is fixed.
+		return this.orch.getPersistenceModel().update(query, { $set: { data } }, { upsert }) as Promise<unknown> as Promise<string>;
```

Comment:
cast from UpdateResult to string.

Reply:
https://github.com/RocketChat/Rocket.Chat/pull/31757/files/3238d4dee1e327cc1cc960092cfdc27d6b800390#r1492815143